nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ValidateJson.java

```diff
@@ -0,0 +1,202 @@
+        this.allowableValues = Collections.unmodifiableList(allowableValues);
+
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    private ObjectMapper mapper = new ObjectMapper();
+    private VersionFlag schemaVersion;
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        // Set JSON schema version to use from processor property
+        this.schemaVersion = VersionFlag.V201909;
+        if (context.getProperty(SCHEMA_VERSION).getValue() == SCHEMA_VERSION_4.getValue()) {
+            this.schemaVersion = VersionFlag.V4;
+        }
+        if (context.getProperty(SCHEMA_VERSION).getValue() == SCHEMA_VERSION_6.getValue()) {
+            this.schemaVersion = VersionFlag.V6;
+        }
+        if (context.getProperty(SCHEMA_VERSION).getValue() == SCHEMA_VERSION_7.getValue()) {
+            this.schemaVersion = VersionFlag.V7;
+        }
+        if (context.getProperty(SCHEMA_VERSION).getValue() == SCHEMA_VERSION_V201909.getValue()) {
+            this.schemaVersion = VersionFlag.V201909;
+        }
+    }
+
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if ( flowFile == null ) {
+            return;
+        }
+
+        try (InputStream in = session.read(flowFile)) {
+            // Read in flowFile inputstream, and validate against schema
+            JsonNode node = mapper.readTree(in);
+            String schemaText = context.getProperty(SCHEMA_TEXT).evaluateAttributeExpressions().getValue();
+            JsonSchemaFactory factory = JsonSchemaFactory.getInstance(schemaVersion);
+            JsonSchema schema = factory.getSchema(schemaText);
+            Set<ValidationMessage> errors = schema.validate(node);
+
+            if (errors.size() > 0) {
+                // Schema checks failed
+                flowFile = session.putAttribute(flowFile, ERROR_ATTRIBUTE_KEY, errors.toString());
+                this.getLogger().info("Found {} to be invalid when validated against schema; routing to 'invalid'", flowFile);
+                session.getProvenanceReporter().route(flowFile, REL_INVALID);
+                session.transfer(flowFile, REL_INVALID);
+            } else {
+                // Schema check passed
+                this.getLogger().debug("Successfully validated {} against schema; routing to 'valid'", flowFile);
+                session.getProvenanceReporter().route(flowFile, REL_VALID);
+                session.transfer(flowFile, REL_VALID);
+            }
+
+        } catch (IOException ioe) {
+            // Failed to read flowFile
+            this.getLogger().error("Failed to process {}, routing to 'failure' details:", flowFile, ioe);
```

Comment:
The trailing `:` character should be removed, and the message should be adjusted:
```suggestion
            this.getLogger().error("JSON processing failed {}", flowFile, ioe);
```